### ISSUE TITLE
Get event when calling the onChange function

### ISSUE THREAD

**Issue by nklswbr** [cchanxzy/react-currency-input-field#102]
It would be great if the onChange function would provide the event that gets triggered like the default onChange function in react does. The event type should be `React.ChangeEvent<HTMLInputElement>`

**Comment by cchanxzy** [comment]
Hi @nklswbr, this is something I have been thinking about as well. It would be a breaking change though, so it's something I want to get right first time.

I think there are a couple of options:

1. Change the `onChange` function so it is called with the event, and also the value eg. `onChange((value: string, event: ChangeEvent<HTMLInputElement>`

2. Change `onChange` to `onValueChange`, so the original `onChange` won't be overwritten, and `onValueChange ` is how you get the unformatted value

I'm still trying to decide which is the better option and would be more future proof.

Luckily, I am planning a major release soon (probably in two weeks), so I might try and get this change in too.

Let me know if you have an opinion :)

**Comment by nklswbr** [comment]
I would prefer the second option because every React developer is used to the original onChange function and expect to retrieve an `ChangeEvent<HTMLInputElement>` event. I would really appreciate if this would be part of the next release.

**Comment by cchanxzy** [comment]
Hi, just as an update, I have released [v3.0.0-beta.7](https://www.npmjs.com/package/react-currency-input-field/v/3.0.0-beta.7) which includes the breaking change: `onChange` has been renamed to `onValueChange`

```js
import CurrencyInput from 'react-currency-input-field';

<CurrencyInput intlConfig={{ locale: 'en-IN', currency: 'INR' }} />;
```

It's still in the testing phase, but I hope this helps. 

I'm going to close the issue, but feel free to re-open if you need to.
